sys/net/gnrc/network_layer/ipv6/blacklist/gnrc_ipv6_blacklist.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author Martin landsmann <martin.landsmann@haw-hamburg.de>
+ */
+
+#include "bitfield.h"
+
+#include "net/gnrc/ipv6/blacklist.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+ipv6_addr_t gnrc_ipv6_blacklist[GNRC_IPV6_BLACKLIST_SIZE];
+BITFIELD(gnrc_ipv6_blacklist_set, GNRC_IPV6_BLACKLIST_SIZE);
+
+#if ENABLE_DEBUG
+static char addr_str[IPV6_ADDR_MAX_STR_LEN];
+#endif
+
+int gnrc_ipv6_blacklist_add(const ipv6_addr_t *addr)
+{
+    for (int i = 0; i < GNRC_IPV6_BLACKLIST_SIZE; i++) {
+        if (!bf_isset(gnrc_ipv6_blacklist_set, i)) {
+            bf_set(gnrc_ipv6_blacklist_set, i);
+            gnrc_ipv6_blacklist[i].u64[0].u64 = addr->u64[0].u64;
```

Comment:
```
~/r/R/e/gnrc_networking > make info-buildsizes-diff OLDBIN=blacklist-bin NEWBIN=blacklist2-bin
text    data    bss     dec     BOARD/BINDIRBASE

-8      0       0       -8      arduino-due
58184   224     20032   78440   blacklist-bin
58176   224     20032   78432   blacklist2-bin

-146    0       0       -146    arduino-mega2560
81976   11044   13881   106901  blacklist-bin
81830   11044   13881   106755  blacklist2-bin

-120    0       0       -120    avsextrem
133416  2332    95965   231713  blacklist-bin
133296  2332    95965   231593  blacklist2-bin

-8      0       0       -8      cc2538dk
58124   220     20008   78352   blacklist-bin
58116   220     20008   78344   blacklist2-bin

0       0       0       0       ek-lm4f120xl
58460   220     20008   78688   blacklist-bin
58460   220     20008   78688   blacklist2-bin

-16     0       0       -16     f4vi1
60064   224     20136   80424   blacklist-bin
60048   224     20136   80408   blacklist2-bin

-8      0       0       -8      fox
73732   220     23096   97048   blacklist-bin
73724   220     23096   97040   blacklist2-bin

-16     0       0       -16     frdm-k64f
60212   1248    20000   81460   blacklist-bin
60196   1248    20000   81444   blacklist2-bin

-8      0       0       -8      iotlab-m3
73460   220     23096   96776   blacklist-bin
73452   220     23096   96768   blacklist2-bin

-8      0       0       -8      limifrog-v1
59312   220     20136   79668   blacklist-bin
59304   220     20136   79660   blacklist2-bin

-8      0       0       -8      mbed_lpc1768
57784   220     20008   78012   blacklist-bin
57776   220     20008   78004   blacklist2-bin

-112    0       0       -112    msba2
160568  2332    95965   258865  blacklist-bin
160456  2332    95965   258753  blacklist2-bin

-16     0       0       -16     msbiot
60376   224     20168   80768   blacklist-bin
60360   224     20168   80752   blacklist2-bin

-16     0       0       -16     mulle
78484   1284    23376   103144  blacklist-bin
78468   1284    23376   103128  blacklist2-bin

-140    0       0       -140    nucleo-f091
60940   220     20008   81168   blacklist-bin
60800   220     20008   81028   blacklist2-bin

0       0       0       0       nucleo-f303
58316   220     20016   78552   blacklist-bin
58316   220     20016   78552   blacklist2-bin

-16     0       0       -16     nucleo-f401
60064   224     20136   80424   blacklist-bin
60048   224     20136   80408   blacklist2-bin

-8      0       0       -8      nucleo-l1
59196   220     20128   79544   blacklist-bin
59188   220     20128   79536   blacklist2-bin

-8      0       0       -8      openmote-cc2538
58032   220     20008   78260   blacklist-bin
58024   220     20008   78252   blacklist2-bin

0       0       0       0       pba-d-01-kw2x
75688   1248    23440   100376  blacklist-bin
75688   1248    23440   100376  blacklist2-bin

-120    0       0       -120    pttu
133624  2332    95965   231921  blacklist-bin
133504  2332    95965   231801  blacklist2-bin

-8      0       0       -8      remote
58708   220     20008   78936   blacklist-bin
58700   220     20008   78928   blacklist2-bin

-140    0       0       -140    saml21-xpro
61136   220     20128   81484   blacklist-bin
60996   220     20128   81344   blacklist2-bin

-140    0       0       -140    samr21-xpro
78092   220     23112   101424  blacklist-bin
77952   220     23112   101284  blacklist2-bin

0       0       0       0       slwstk6220a
57960   220     20136   78316   blacklist-bin
57960   220     20136   78316   blacklist2-bin

-16     0       0       -16     stm32f3discovery
58332   220     20016   78568   blacklist-bin
58316   220     20016   78552   blacklist2-bin

-16     0       0       -16     stm32f4discovery
60308   224     20152   80684   blacklist-bin
60292   224     20152   80668   blacklist2-bin

-8      0       0       -8      udoo
58184   224     20032   78440   blacklist-bin
58176   224     20032   78432   blacklist2-bin
```

`blacklist2-bin` was compiled with the following diff:
```diff
diff --git a/examples/gnrc_networking/Makefile b/examples/gnrc_networking/Makefile
index b73bb4d..fae13c5 100644
--- a/examples/gnrc_networking/Makefile
+++ b/examples/gnrc_networking/Makefile
@@ -29,6 +29,7 @@ USEMODULE += gnrc_icmpv6_echo
 USEMODULE += shell
 USEMODULE += shell_commands
 USEMODULE += ps
+USEMODULE += gnrc_ipv6_blacklist

 # Comment this out to disable code in RIOT that does safety checking
 # which is not needed in a production environment but helps in the
diff --git a/sys/net/gnrc/network_layer/ipv6/blacklist/gnrc_ipv6_blacklist.c b/sys/net/gnrc/network_layer/ipv6/blacklist/gnrc_ipv6_blac
index da4026e..3256ed2 100644
--- a/sys/net/gnrc/network_layer/ipv6/blacklist/gnrc_ipv6_blacklist.c
+++ b/sys/net/gnrc/network_layer/ipv6/blacklist/gnrc_ipv6_blacklist.c
@@ -15,6 +15,7 @@
  */

 #include "bitfield.h"
+#include <string.h>

 #include "net/gnrc/ipv6/blacklist.h"

@@ -33,8 +34,7 @@ int gnrc_ipv6_blacklist_add(const ipv6_addr_t *addr)
     for (int i = 0; i < GNRC_IPV6_BLACKLIST_SIZE; i++) {
         if (!bf_isset(gnrc_ipv6_blacklist_set, i)) {
             bf_set(gnrc_ipv6_blacklist_set, i);
-            gnrc_ipv6_blacklist[i].u64[0].u64 = addr->u64[0].u64;
-            gnrc_ipv6_blacklist[i].u64[1].u64 = addr->u64[1].u64;
+            memcpy(&gnrc_ipv6_blacklist[i], addr, sizeof(ipv6_addr_t));
             DEBUG("IPv6 blacklist: blacklisted %s\n",
                   ipv6_addr_to_str(addr_str, addr, sizeof(addr_str)));
             return 0;
```

memcpy seems to be more efficient on most platforms than assigning two `uint_64`'s